internal/incident/db_types.go

```diff
@@ -85,6 +89,26 @@ func (h *HistoryRow) TableName() string {
 	return "incident_history"
 }
 
+// Sync persists the current state of this history to the database.
+// You can set the last argument to true if you want to retrieve and set the just inserted history ID.
+// Returns error when failed to execute the query.
+func (h *HistoryRow) Sync(ctx context.Context, db *icingadb.DB, tx *sqlx.Tx, fetchID bool) error {
+	stmt := utils.BuildInsertStmtWithout(db, h, "id")
+	if fetchID {
+		historyId, err := utils.InsertAndFetchId(ctx, tx, stmt, h)
+		if err != nil {
+			return err
+		}
+
+		h.ID = historyId
+
+		return nil
+	}
+
+	_, err := tx.NamedExecContext(ctx, stmt, h)
+	return err
+}
```

Comment:
I'd just assume `fetchID = true` here and remove the parameter. It's not like that would be much more expensive for the database, so this mostly just increases code complexity here.